src/main/java/org/testng/internal/TestResult.java

```diff
@@ -230,7 +231,8 @@ private static String toString(int status) {
       case SKIP: return "SKIP";
       case SUCCESS_PERCENTAGE_FAILURE: return "SUCCESS WITHIN PERCENTAGE";
       case STARTED: return "STARTED";
-      default: throw new RuntimeException();
+      case CREATED: return "CREATED";
+      default: throw new TestNGException("Encountered an un-defined test status of [" + status + "].");

```

Comment:
i'd let default to return something `un-defined test status of [" + status + "]`, the point is `toString()` should be as quiet as possible.
however, now that the case branches cover all the status values, so i'm fine with this.